### PR TITLE
Add test case for network timeout not during fetch call

### PR DESCRIPTION
**Pull request by jdelbick**
## Description

Adds test case for network timeouts outside of the fetch call/ retry check: https://github.com/adobe/node-fetch-retry/blob/master/index.js#L245

This is mostly so every time we have a new PR, that adds some code (even if covered), it doesn't tell us we lost code coverage: https://github.com/adobe/node-fetch-retry/pull/90/checks?check_run_id=6180692734
before:
```
----------|---------|----------|---------|---------|-------------------
File      | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s 
----------|---------|----------|---------|---------|-------------------
All files |    98.9 |     98.2 |     100 |   98.88 |                   
 index.js |    98.9 |     98.2 |     100 |   98.88 | 245               
----------|---------|----------|---------|---------|-------------------
```
after:
```
----------|---------|----------|---------|---------|-------------------
File      | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s 
----------|---------|----------|---------|---------|-------------------
All files |     100 |     98.2 |     100 |     100 |                   
 index.js |     100 |     98.2 |     100 |     100 | 117,240           
----------|---------|----------|---------|---------|-------------------
```

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: test/index.test.js [hunk 1]
```diff
@@ -937,6 +939,23 @@ describe('test fetch retry on http errors (throw exceptions)', () => {
...
+        const getTimeRemainingStub = sinon.stub(getTimeRemainingObject, 'getTimeRemaining').callsFake(() => {
+            console.log("Mocked isResponseTimedOut");
+            return -1;
+        });
+        rewiredFetchRetry.__set__('getTimeRemaining', getTimeRemainingStub);
+        try {
+            await rewiredFetchRetry(`${FAKE_BASE_URL}${FAKE_PATH}`, { method: 'GET' });
+            assert.fail("Should have thrown an error!");
+        } catch (e) {
+            assert(e.message.includes("network timeout"));
```
Comment (tmathern) [inline test/index.test.js:955]: Ack - I know this is the style we did those tests. But a good change would also to check the expected exception using something like `assert.throws(...)`.

**Review by alexkli — COMMENTED** [review]
Note that the current error message `network timeout` (which is tested here0 is confusing and we should change it: #81

Also the test seems to be very implementation specific (and uses sinon rewiring tricks). It would be better if this could be tested black box, so we can refactor things without breaking this test.

**Comment by jdelbick** [comment]
@alexkli good point
I could go one level up and mock `isResponseTimedOut`. Any other ideas?

**Comment by alexkli** [comment]
I notice that the existing tests already do a lot of rewiring of internals... IMO that is an antipattern as all the tests are baked with the implementation. Unit tests should use the API of the module/library, which in this case is the `fetch()` call with the retryOptions support this library adds.

My optimistic assumption is that we could mock the HTTP requests using `nock` and have them provide the error responses or timeout delay that we need (we do similar things with nock in other places, such as [node-openwhisk-newrelic](https://github.com/adobe/node-openwhisk-newrelic)). And then do normal `fetch()` calls and inspect the result or error responses.

For this in particular case: I wonder why we even need that logic. It seems to happen in weird situations anyway (see #81 - I still don't fully understand what happened there). It seems that we don't need it at all but should check this in a different place. After one request failed, we are sleeping here

https://github.com/adobe/node-fetch-retry/blob/master/index.js#L239-L242

but we should check here if after the sleep time we will exceed maxRetry. So we could fail right away...

We probably still need the check to be sure we don't exceed max retry, but maybe we don't need to test it.

The code is a bit hard to follow. Also I wonder why we are using raw Promise here where we could be using async/await instead: https://github.com/adobe/node-fetch-retry/blob/master/index.js#L204-L205

**Comment by alexkli** [comment]
FYI - simplified the logic using async/await in #93.

**Comment by jdelbick** [comment]
Closing and will re-do after/during https://github.com/adobe/node-fetch-retry/issues/81
